pytensor/tensor/sharedvar.py

```diff
@@ -9,6 +9,16 @@
 from pytensor.tensor.variable import _tensor_py_operators
 
 
+def __getattr__(name):
+    if name == "ScalarSharedVariable":
+        warnings.warn(
+            "The class `ScalarSharedVariable` has been deprecated. "
+            "Use `TensorSharedVariable` instead and check for `ndim==0`.",
+            FutureWarning,
+        )
+        return TensorSharedVariable
+
+
```

Comment:
very clever!

Reply:
Oops not clear, enough have to fallback to AttributeError